modules/extensions-support/src/main/java/org/mule/runtime/module/extension/api/runtime/privileged/ChildContextChain.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.api.runtime.privileged;
+
+import org.mule.runtime.extension.api.runtime.operation.Result;
+import org.mule.runtime.extension.api.runtime.route.Chain;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+/**
+ * Extension of @{link Chain} that adds the feature to execute the chain with a given correlation id.
```

Comment:
`with a given correlation id ` -> `with a child context` (and link to the java doc of the new `child` method from `DefaultEventContext`)